src/elm/aggregate.coffee

```diff
@@ -227,13 +234,54 @@ module.exports.AllTrue = class AllTrue extends AggregateExpression
     super
 
   exec: (ctx) ->
-    args =@source.execute(ctx)
-    allTrue(args)
+    items = @source.execute(ctx)
+    allTrue(items)
 
 module.exports.AnyTrue = class AnyTrue extends AggregateExpression
   constructor:(json) ->
     super
 
   exec: (ctx) ->
-    args = @source.execute(ctx)
-    anyTrue(args)
+    items = @source.execute(ctx)
+    anyTrue(items)
+
+processQuantities = (values) ->
+  values = removeNulls(values)
+  if hasOnlyQuantities(values)
+    values = convertAllUnits(values)
+  else if hasSomeQuantities(values)
+    throw new Exception("Cannot perform aggregate operations on mixed values of Quantities and non Quantities")
+  else
+    values
+
+getValuesFromQuantities = (quantities) ->
+  quantities.map (quantity) -> quantity.value
+
+hasOnlyQuantities = (arr) ->
+  arr.every (x) -> x.isQuantity
+
+hasSomeQuantities = (arr) ->
+  arr.some (x) -> x.isQuantity
+
+convertAllUnits = (arr) ->
+  # convert all quantities in array to match the unit of the first item
+  converted = []
+  for quantity in arr
+    converted.push(quantity.convertUnit(arr[0].unit))
+  converted
+
+setQuantityUnit = (quantity, arr) ->
```

Comment:
As best I can tell, this function isn't used anywhere, is it?

Reply:
It's not!  It was ugly so I refactored the code to not need it anymore, then forgot to remove it.  Will remove